### DISCUSSION TITLE
Gaps in axis range

### DISCUSSION THREAD

**Discussion by osaidmakhalfeh** [GSI-CS-CO/chart-fx#396] [Ideas]
Hello Dears,

I'm very happy while using and contributing to this library, thank you all.

I'm working on an application that reads data from a hardware sensor and re-fill the data as bar charts every two seconds. I have two problems and want to discuss it with you:

1) sometimes some bars not rendered!, I'm not sure why?, to be sure with that point I entered the same data array by code "static code" and unfortunately  I'm getting the same result which is-> the chart does not render all bars.
here is my snapshot of the result and its input:
![bugHugeRane4](https://user-images.githubusercontent.com/44381968/94337219-89a95a80-fff1-11ea-92ea-5d89e5dab2fe.PNG)
As you see here, I should get 17 bars instead of 14!

2) Because i have a good range between yValues. I'm not sure if I can jump the yAxis values to make the user feel that s/he can compare between bars easily! 
   I tried to simulate my expected result as you see on the following image (on the right), do not know if someone has an idea!
![bugHugeRane1](https://user-images.githubusercontent.com/44381968/94337384-ad20d500-fff2-11ea-9a65-0b2efbdb8dc0.PNG)

**Comment by osaidmakhalfeh** [comment]
@RalphSteinhagen 
I'm very thankful to you for your helping. 

I saw your reply on this issue:
https://github.com/GSI-CS-CO/chart-fx/issues/265

Surely you and I notice that this is the same problem, I tried the solution you gave to me before, 
"ErrorDataSetRenderer::setPointReduction(boolean)"

but unfortunately, I'm getting the same result 3 of bars get drawn only! 
I tried to put fake data, like "dataSet_Default.add(21, 0);", and that solves the issue, bar number 4 got drawn, but at the running time there will be new data set, so my dummy data will not work!

this is a snapshot of my test:
![xxx](https://user-images.githubusercontent.com/44381968/94403906-cb174280-0176-11eb-91ab-95e668a4e7b3.png)

A lot of thanks

**Comment by wirew0rm** [comment]
hey @osaidmakhalfeh,

for future issue reports, please try not to mix multiple topics into one issue, as it makes it a lot harder to follow and manage the disucussion.
I will only answer the second question on this issue, as you already said that the other one is closely related to #265, and should be further discussed there. I also changed the title of the issue to reflect this.

The feature you want to use is currently not implemented in chartfx. You would need to implement your own Axis, which can define arbitrary ticks and coordinate transforms, and also implement the visualization of the axis gap. I suggest to look at [LinearAxis](https://github.com/GSI-CS-CO/chart-fx/blob/master/chartfx-chart/src/main/java/de/gsi/chart/axes/spi/LinearAxis.java) or [DefaultNumericAxis](https://github.com/GSI-CS-CO/chart-fx/blob/master/chartfx-chart/src/main/java/de/gsi/chart/axes/spi/DefaultNumericAxis.java).

You would also have to come up with a way to either supply the ranges of the axis gaps to the plot or to determine them automatically (computationally expensive, axes usually do not have access to the dataset).

Alternatively think about other ways of displaying your data (logarithmic plot or apply some normalisation before).

I'll keep this issue open in case you want to give it a try and implement this and have further questions
